### PR TITLE
added legacy credential model

### PR DESCRIPTION
**Pull request by elfeto**
Added the model for credentialing.
Also it checks when a account is created and a email is added to see if the person was credentialed in the past.

### CODE REVIEW

File: physionet-django/user/templates/user/edit_credentialing.html [hunk 1]
```diff
@@ -17,6 +17,7 @@ <h1>PhysioNet Credentialing</h1>
   <p><i class="fa fa-clock"></i> Your credentialing application was submitted on {{ current_application.application_datetime }}. We will respond shortly.</p>
 {% else %}
   <p>Your account is not credentialed. You may <a href="{% url 'credential_application' %}">apply for access</a>.</p>
+  <p><strong>If you have already been credentialed</strong>, please add the approved email to your profile.</p>
```
Comment (cx1111) [inline physionet-django/user/templates/user/edit_credentialing.html]: "If your account on the old PhysioNet site is already credentialed, please add the email address from the old site to this account. Your credentialed status will be automatically transferred when your email is verified."

I think we have to use the word 'old' here. Make it 100% clear.

**Comment by cx1111** [comment]
Great feature. First thing to do is to pull the common code into a `check_legacy_credentials` function and call it from `activate_user` and `verify_email`, not `register`.

File: physionet-django/user/views.py [hunk 2]
```diff
@@ -286,6 +296,17 @@ def verify_email(request, uidb64, token):
             associated_email.verification_date = timezone.now()
             associated_email.is_verified = True
             associated_email.save()
+
```
Comment (tompollard) [inline physionet-django/user/views.py]: Can you avoid this duplication with something like a `legacycredentialed(email)` function? Might be nice to separate it out.

**Comment by tompollard** [comment]
This is excellent! Thought it was going to be super complicated to work out...big step towards being able to switch to the new system.

**Comment by elfeto** [comment]
- Made the change to avoid the duplication of the credentialing code
- Fixed it so that a person can only use this feature once.

I will do another pull request adding some tests

**Pull request merged**